src/tools/miri/tests/fail/stacked_borrows/drop_in_place.rs

```diff
@@ -0,0 +1,21 @@
+#[repr(transparent)]
+struct HasDrop(u8);
```

Comment:
This test already [has UB with current Miri](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8bce42233ba3b7b16349cdf14ab86204), so it doesn't really test your changes.

Reply:
[Here is a version](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5c230abdc45b4c55d4cdca8233cd7c97) that currently passes, which hopefully is declared UB with your patch.

Reply:
And [here is another test](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=483452b2a2066b6e15f50c7eeecf9c25).

Reply:
The semantics we have right now are that `drop_in_place` is still a nop if the type genuinely has no drop glue at all. I've adjusted the second test accordingly.

Reply:
Oh, is that compatible with @pcwalton's plans? And why is that, can't we make it always retag?

The second test was quite deliberately designed to hit the case of a type that does not need dropping.

Reply:
Yes, it is, since we never codegen nop `drop_in_place` (either for direct function calls or `TerminatorKind::Drop`). This is the result of the check [here](https://github.com/rust-lang/rust/blob/07e664c41ae5679b912ba9d1f7364ebaac8e51b6/compiler/rustc_ty_utils/src/instance.rs#L60-L78). Although codegen also separately [ignores these](https://github.com/rust-lang/rust/blob/1eb62b1235fd77200e6bd967d70e83c0f2497233/compiler/rustc_codegen_ssa/src/mir/block.rs#L768-L772). That should make sure everything is compatible, but I'll add a test ensuring it stays that way

Reply:
> and as best I can tell from reading the LLVM docs putting noalias on an unused argument probably does nothing (although who knows).

There's still `dereferenceable aligned nonnull` though, right?